packages/block-editor/src/components/global-styles/background-panel.js

```diff
@@ -720,44 +939,84 @@ export default function BackgroundPanel( {
-									resetBackground();
-								} }
-							/>
-							<BackgroundSizeControls
-								onChange={ onChange }
-								panelId={ panelId }
-								style={ value }
-								defaultValues={ defaultValues }
-								inheritedValue={ inheritedValue }
-								themeFileURIs={ themeFileURIs }
-							/>
-						</VStack>
-					</BackgroundControlsPanel>
-				) : (
-					<BackgroundImageControls
-						onChange={ onChange }
-						style={ value }
-						inheritedValue={ inheritedValue }
-						themeFileURIs={ themeFileURIs }
-					/>
-				) }
+				<>
+					{ shouldShowBackgroundImageControls ? (
+						<BackgroundControlsPanel
+							label={ title }
+							filename={ title }
+							url={ getResolvedThemeFilePath(
+								url,
+								themeFileURIs
+							) }
+							onToggle={ setIsDropDownOpen }
+							hasImageValue={ hasImageValue }
+						>
+							<VStack spacing={ 3 } className="single-column">
+								<BackgroundImageControls
+									onChange={ onChange }
+									style={ value }
+									inheritedValue={ inheritedValue }
+									themeFileURIs={ themeFileURIs }
+									displayInPanel
+									onRemoveImage={ () => {
+										setIsDropDownOpen( false );
+										resetBackground();
+									} }
+								/>
+								<BackgroundSizeControls
+									onChange={ onChange }
+									panelId={ panelId }
+									style={ value }
+									defaultValues={ defaultValues }
+									inheritedValue={ inheritedValue }
+									themeFileURIs={ themeFileURIs }
+								/>
+							</VStack>
+						</BackgroundControlsPanel>
+					) : (
+						<BackgroundImageControls
+							onChange={ onChange }
+							style={ value }
+							inheritedValue={ inheritedValue }
+							themeFileURIs={ themeFileURIs }
+						/>
+					) }
+					{ showBackgroundPanel && path === '/background' && (
+						<ColorPanelDropdown
+							key="background"
+							label={ __( 'Color' ) }
+							hasValue={ hasBackground }
+							resetValue={ resetBackgroundColor }
+							isShownByDefault={ defaultControls.background }
+							indicators={ [ gradient ?? backgroundColor ] }
+							tabs={ [
+								hasSolidColors && {
+									key: 'background',
+									label: __( 'Background' ),
+									inheritedValue: backgroundColor,
+									setValue: setBackgroundColor,
+									userValue: userBackgroundColor,
+								},
+								hasGradientColors && {
+									key: 'gradient',
+									label: __( 'Gradient' ),
+									inheritedValue: gradient,
+									setValue: setGradient,
+									userValue: userGradient,
+									isGradient: true,
+								},
+							].filter( Boolean ) }
+							colorGradientControlSettings={ {
+								colors,
+								disableCustomColors: ! areCustomSolidsEnabled,
+								gradients,
+								disableCustomGradients:
+									! areCustomGradientsEnabled,
+							} }
+							panelId={ panelId }
+						/>
+					) }
+				</>
 			</div>
 
 			{ /* Dummy ToolsPanel items, so we can control what's in the dropdown popover */ }
```

Comment:
I think this "Dummy ToolsPanel" hack is causing the item order in the dropdown to be out of sync with the actual items. Not immediately sure why we need to hack it like this — ideally it shouldn't be a hack, but we could at least move it up the file so the order is correct.

<img src="https://github.com/user-attachments/assets/5099060a-70c4-4fc6-beaf-a15dcb50301a" alt="ToolsPanel dropdown out of order" width="556">

Reply:
I was reporting the same thing :)

Reply:
I [added dummy tools panel item](https://github.com/WordPress/gutenberg/pull/60151#issuecomment-2024403013) so that the toolspanel ellipsis menu would play nicely, at least semi-nicely with the popover.

It was unclear whether we wanted to preserve the ToolPanel functionality of being able to select "default" controls, and also toggle non-default block support controls.

You can see this in action with many of the Typography controls.

https://github.com/user-attachments/assets/de174b82-66ce-48b4-840f-94d931d336db

Tools panel items need to be rendered inside a ToolsPanel component in order to be registered. Because the controls are in the popover, toggling these non-default controls didn't work while the popover was closed.

So essentially it was a bit of a hack, but one that would leave open the possibility of allowing toggling. It  was introduced before I [moved the controls into the popover](https://github.com/WordPress/gutenberg/pull/60151).

In my mind was the [ongoing work](https://github.com/WordPress/gutenberg/pull/60739) to refactor `color.gradient` to share the `background-image` CSS property, and possibility of applying position/size/repeat values to gradient and not just images.

**However...**

Now that you've brought it up, I reckon we could reconsider all this and let the background controls all live as one control group, with no sub-style toggling. 

Especially since they're now in a popover and space isn't such an issue, but also I think it would be annoying to have to toggle.

Whenever the gradient/image consolidation takes place, we can revisit the controls. 

This PR is big enough, I'll throw up a smaller one 😄

Reply:
PR here: https://github.com/WordPress/gutenberg/pull/64242